`make generate` can not be run more than once

### What happened?

Running `make generate` more than once results in the following error:
```
20:50:21 [ .. ] patching generated CRDs
mkdir: /Users/jared/dev/go/src/github.com/crossplane/crossplane/.work/patch: File exists
make[1]: *** [crds.patch] Error 1
make: *** [generate] Error 2
```

This also leaves a side effect of the `cluster/crds/pkg.crossplane.io_deploymentruntimeconfigs.yaml` file being modified with the following diff:

```diff
diff --git a/cluster/crds/pkg.crossplane.io_deploymentruntimeconfigs.yaml b/cluster/crds/pkg.crossplane.io_deploymentruntimeconfigs.yaml
index 35a599ab7..f876d0e55 100644
--- a/cluster/crds/pkg.crossplane.io_deploymentruntimeconfigs.yaml
+++ b/cluster/crds/pkg.crossplane.io_deploymentruntimeconfigs.yaml
@@ -4211,8 +4211,6 @@ spec:
                                     ip:
                                       description: IP address of the host file entry.
                                       type: string
-                                  required:
-                                  - ip
                                   type: object
                                 type: array
                                 x-kubernetes-list-map-keys:
@@ -4267,8 +4265,6 @@ spec:
                                         More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
                                         TODO: Add other useful fields. apiVersion, kind, uid?
                                       type: string
-                                  required:
-                                  - name
                                   type: object
                                   x-kubernetes-map-type: atomic
                                 type: array
```

This looks related to recently added build logic in https://github.com/crossplane/crossplane/commit/355ba7a44b4b23240357fd3e3447d2d3239ffd44

### How can we reproduce it?

Run `make generate` two times in a row.

Workaround: `rm -fr .work/patch` and then run `make generate` again.

### What environment did it happen in?
Crossplane version: `v1.17.0-rc.0.120.g914b83a3c`, corresponding to  https://github.com/crossplane/crossplane/commit/914b83a3c65e73aef00c9b96aafd57be01dbe98e 

<!--
Include at least the version or commit of Crossplane you were running. Consider
also including your:

* Cloud provider or hardware configuration
* Kubernetes version (use `kubectl version`)
* Kubernetes distribution (e.g. Tectonic, GKE, OpenShift)
* OS (e.g. from /etc/os-release)
* Kernel (e.g. `uname -a`)
-->

--- a/Makefile
+++ b/Makefile
@@ -86,7 +86,7 @@ CRD_PATCH_DIR = cluster/crd-patches
 # See patch files for details.
 crds.patch: $(KUBECTL)
 	@$(INFO) patching generated CRDs
-	@mkdir $(WORK_DIR)/patch
+	@mkdir -p $(WORK_DIR)/patch
 	@$(KUBECTL) patch --local --type=json -f $(CRD_DIR)/pkg.crossplane.io_deploymentruntimeconfigs.yaml --patch-file $(CRD_PATCH_DIR)/pkg.crossplane.io_deploymentruntimeconfigs.yaml -o yaml > $(WORK_DIR)/patch/pkg.crossplane.io_deploymentruntimeconfigs.yaml
 	@mv $(WORK_DIR)/patch/pkg.crossplane.io_deploymentruntimeconfigs.yaml $(CRD_DIR)/pkg.crossplane.io_deploymentruntimeconfigs.yaml
 	@$(OK) patched generated CRDs

